Labs/module-1_labs/03_mysql/lab_sql-extras/olist_data/exercises/lab-2-olist.md

```diff
@@ -0,0 +1,173 @@
+![IronHack Logo](https://s3-eu-west-1.amazonaws.com/ih-materials/uploads/upload_d5c5793015fec3be28a63c4fa3dd4d55.png)
+
+# Lab | SQL
+
+## Introduction
+
+Run your first SQL commands! In this lab, we will practice selecting and projecting data. You can finish all questions with only those clauses:
+- `SELECT`
+- `SELECT DISTINCT`
+- `FROM`
+- `WHERE`
+- `ORDER BY`
+- `GROUP BY`
+- `LIMIT`
+
+and aggregate functions:
+- `COUNT`
+- `SUM`
+- `MIN`
+- `MAX`
+
+Please submit your solutions in a text file `lab-2-olist-solutions.sql`.
+
+### 1. From the `marketing_qualified_leads` table, find the earliest and latest first_contact_date.
+
+Result:
+```
+2017-06-14 and 2018-05-31
+```
+Solution:
+```sql
+SELECT
+    MIN(first_contact_date),
+    MAX(first_contact_date)
+FROM olist.marketing_qualified_leads;
+```
+
+### 2. From the `marketing_qualified_leads` table, find the top 3 most frequent origin types for all first_contact_date entries in 2018.
+Result:
+```
+organic_search, paid_search and social
+```
+Solution:
+```sql
+SELECT
+    COUNT(first_contact_date),
+    origin
+FROM olist.marketing_qualified_leads
+WHERE first_contact_date >= 2018-01-01
```

Comment:
cool! didn't know you don't have to wrap the date into quotes

Reply:
may need to set upper threshold as well instead of relying on data to be prior to 2019